miio/roidmivacuum_miot.py

```diff
@@ -0,0 +1,590 @@
+            _LOGGER.error("Unknown PathMode (%s)", self.data["path_mode"])
+            return PathMode.Unknown
+
+    @property
+    def mop_present(self) -> bool:
+        return self.data["mop_present"]
+
+    @property
+    def work_station_freq(self) -> int:
+        return self.data["work_station_freq"]
+
+    @property
+    def timing(self) -> str:
+        return self.data["timing"]
+
+    @property
+    def clean_area(self) -> int:
+        return self.data["clean_area"]
+
+    @property
+    def uid(self) -> int:
+        return self.data["uid"]
+
+    @property
+    def auto_boost(self) -> int:
+        return self.data["auto_boost"]
+
+    def parseForbidMode(self, val):
+        def secToClock(val):
+            hour = math.floor(val / 3600)
+            minut = math.floor((val - hour * 3600) / 60)
+            return "{}:{:02}".format(hour, minut)
+
+        asDict = json.loads(val)
+        active = bool(asDict["time"][2])
+        begin = secToClock(asDict["time"][0])
+        end = secToClock(asDict["time"][1])
+        return json.dumps(
+            {"enabled": active, "begin": begin, "end": end, "tz": asDict["tz"]}
+        )
+
+    @property
+    def forbid_mode(self) -> int:
+        # Example data: {"time":[75600,21600,1],"tz":2,"tzs":7200}
+        return self.parseForbidMode(self.data["forbid_mode"])
+
+    @property
+    def water_level(self) -> WaterLevel:
+        try:
+            return WaterLevel(self.data["water_level"])
+        except ValueError:
+            _LOGGER.error("Unknown WaterLevel (%s)", self.data["water_level"])
+            return WaterLevel.Unknown
+
+    # @property
+    # def siid8_13(self) -> int:
+    #     return self.data["siid8_13"]
+
+    # @property
+    # def siid8_14(self) -> int:
+    #     return self.data["siid8_14"]
+
+    @property
+    def clean_counts(self) -> int:
+        return self.data["clean_counts"]
+
+    # @property
+    # def siid8_19(self) -> int:
+    #     return self.data["siid8_19"]
+
+    @property
+    def double_clean(self) -> bool:
+        return self.data["double_clean"]
+
+    @property
+    def edge_sweep(self) -> bool:
+        return self.data["edge_sweep"]
+
+    @property
+    def led_switch(self) -> bool:
+        return self.data["led_switch"]
+
+    @property
+    def lidar_collision(self) -> bool:
+        return self.data["lidar_collision"]
+
+    @property
+    def station_key(self) -> bool:
+        return self.data["station_key"]
+
+    @property
+    def station_led(self) -> bool:
+        return self.data["station_led"]
+
+    @property
+    def current_audio(self) -> str:
+        return self.data["current_audio"]
+
+    @property
+    def progress(self) -> str:
+        return self.data["progress"]
+
+    @property
+    def voice_conf(self) -> str:
+        return self.data["voice_conf"]
+
+    @property
+    def switch_status(self) -> SwitchStatus:
+        try:
+            return SwitchStatus(self.data["switch_status"])
```

Comment:
This doesn't currently seem to be queried at all. What does this indicate?

Reply:
I am not sure. It is an Enum and can only have the value "Open". I will remove it from the public interface, and only leave an comment about this in MiotMapping.